x-pack/plugins/security/server/config.ts

```diff
@@ -346,16 +351,23 @@ export function createConfig(
     order: number;
     hasAccessAgreement: boolean;
   }> = [];
+
   for (const [type, providerGroup] of Object.entries(providers)) {
     for (const [name, { enabled, order, accessAgreement }] of Object.entries(providerGroup ?? {})) {
+      const hasProviderSpecificAccessAgreement: boolean = !!accessAgreement?.message;
+      const hasGlobalAccessAgreement: boolean = !!config.accessAgreement?.message;
+
       if (!enabled) {
         delete providerGroup![name];
       } else {
+        const hasAccessAgreement: boolean =
+          hasProviderSpecificAccessAgreement || hasGlobalAccessAgreement;
+
         sortedProviders.push({
           type: type as any,
           name,
           order,
-          hasAccessAgreement: !!accessAgreement?.message,
+          hasAccessAgreement,
```

Comment:
The `security_usage_collector.ts` references that field. Do you think the change would be better there or should I remove it?

Reply:
Ah, I didn't realize that we introduced that field just for usage collector! I'd still make this check directly in the `security_usage_collector.ts` (the place that actually needs these two settings to be merged to a single `accessAgreementEnabled` field), **but** since `hasAccessAgreement` is used solely for the usage collector feel free to pick the option you like the most.

Reply:
Ill make the change in the collector!

Reply:
Done!